tests/test_floyd_warshall.py

```diff
@@ -111,12 +111,15 @@ def test_weighted_numpy_negative_cycle(self):
             (3, 0, -1),
         ])
         dist = retworkx.graph_floyd_warshall_numpy(graph, lambda x: x)
-        expected = numpy.array(
-            [[-6, -7, -8, -13],
-             [-7, -8, -9, -14],
-             [-8, -9, -10, -15],
-             [-13, -14, -15, -20]], dtype=numpy.float64)
-        self.assertTrue(numpy.array_equal(dist, expected))
+        floyd_graph = retworkx.PyGraph()
+        floyd_graph.add_nodes_from(list(range(4)))
+        floyd_graph.add_edges_from(
+            [(i, j, dist[i][j]) for i in range(4) for j in range(4)]
+        )
+        second_dist = retworkx.graph_floyd_warshall_numpy(
+            floyd_graph, lambda x: x
+        )
+        self.assertTrue(numpy.all(second_dist < dist))
```

Comment:
Instead of comparing exact matrices, we compare if there's a negative cycle. If there's a negative cycle and I run Floyd-Warshall's algorithm again, it will give me shorter paths than before

Reply:
Could you just check for negative values in the diagonal of `dist`?

Reply:
Indeed, I switched to that. I think it's a reasonable check as we instantiate the diagonals to 0 at the start.